packages/changeset-form/addon/components/changeset-form/index.ts

```diff
@@ -18,13 +18,12 @@ interface ChangesetFormArgs {
 }
 
 export default class ChangesetForm extends Component<ChangesetFormArgs> {
-  changeset!: BufferedChangeset;
+  @tracked changeset!: BufferedChangeset;
```

Comment:
Interesting approach,

So there's two modes:
1. Manual mode - Passing down a changeset to entirely own the flow
2. Auto Mode - Passing down a model so the component creates a changeset for you

I think to be able to reset using auto mode, we should take into account this.args.model inside the getter to see if we need to create a new `this._changeset`, so we also need to keep track of the previous model

In my humble opinion, I would suggest having only the "manual mode", and encourage the usage of `{{changeset}}` helper for template ergonomics for a simplier mental model

Reply:
You are right about the 2 modes. 

For the reset to work, we do need to have the getter, otherwise, it will not pick up the change as we only set the changeset on the constructor.

I do like the idea of having just one mode, the manual, but I wonder about the validations, as we have that lookupValidations usage...

Reply:
I think we won't need to use lookupValidations if we just support only the manual mode, let me update the PR with a few tweaks I think could make it a bit more flexible

Reply:
I updated the code, what do you think about the form only providing these methods and letting the client pass down the changeset?

Reply:
I like this direction. Let's go with it.

- Let's update the tests
- Ensure we have some tests using validations
- Update the PR description - This will be a breaking change...